### PR TITLE
UCP/MM: Detect memory type in ucp_mem_map()

### PR DESCRIPTION
**Pull request by bureddy**

### CODE REVIEW

File: src/ucp/core/ucp_mm.c [hunk 1]
```diff
@@ -322,9 +322,9 @@ static ucs_status_t ucp_mem_map_common(ucp_context_h context, void *address,
     } else {
         ucs_debug("registering user memory at %p length %zu", address, length);
         memh->alloc_method = UCT_ALLOC_METHOD_LAST;
-        memh->mem_type     = UCS_MEMORY_TYPE_HOST;
         memh->alloc_md     = NULL;
         memh->md_map       = 0;
+        ucp_memory_type_detect_mds(context, address, length, &memh->mem_type);
```
Comment (yosefe) [inline src/ucp/core/ucp_mm.c]: minor: maybe do this before line 323, and add memtype print to the ucs_debug message above?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11617/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 06-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/6d0318ce6f1e409ffab74d8e9dc4f0489d5e31b5/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/10e2663587d0b8339311c5ac1d1cf523922bf72a/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/5bfa0ce22dec440854e66d94d4d9e97068a9dd98/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/a84c6bf81ac7f1bb28fbdd567fda703768b79f3d/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/f6e4a52ccd0b79428d2aa40c020c1376232f9709/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/494719e7509c5d70e854771d7ccedab85978999b/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/94d356545c0f6d8f9058ba819a5a343341ada1a7/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/a89adcb2e6e06ab81bf73f825405e63edf35a7ac/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/4ed77247dd0d9fa128fcb9e75734a3841250cd11/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/419983bd3de453e1373f9778a0d3905a2cd189e1/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/fdbb7dbcea61b780e80c55c5b2629cb6e538f752/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/9c502eee8099f64a333ad6ed59d4217c67eac78b/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/f0b23352da5160dba546f34e35b3170495aa46a9/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/95a574f0d96987b2fc731d10967fbf022dbc5dc4/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/dee37fa739458431adf4194c4348f5fc2742d722/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/bc35776d08c1c596ae57c86a2662aa0935aaef16/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/f83dc4ceb7972b7cc4568aaad7d43bbdeb98909b/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/8cef22f15455bc3d59be64c02bad579b759b5d89/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/76ca548492e4c44df06dd31d2b7a4003b96de9ec/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/5fb7b40ecc1b5abd4eb2f2d347212c45fc3ab5df/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/b0e86dd699f2d975b4cb4cd97cb486307ce09fbc/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/654f08faa9a9dfb49b47e03946a7d7927b045de3/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/1f5e5c61b68ad805df0187ccaae106784b43bd17/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/811640202d0f89e03046f2b5527a56989027c89f/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/5a14c31af7a2f0363b321052ad6ad1217e5f5fe8/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/44ca8e291e9680cbd4d64178bd7c36d5bdd56899/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/7f9cc5bf10c30653b60599d2b44601bfc5041fef/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/09486a27ead077daf02cd8351e5a8ec39bb65312/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/499cbf54069015ee5754ebe1e7cb6d2c/raw/dbce30e729ec632ac9e16e768fd2ab70df4c1cd1/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11624/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 06-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/11c6cd826f99266b68fdb75d1095f204eeeab06a/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/b2bf8816902cec4edf568619c5010c565f906946/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/1eff2dcbecb70170b14bd7f88f17daa2996793c3/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/692dd90adae7ff9cdae42f51bc9aea9e9b138840/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/f5629c37f2a49967191097e1ae6bb85c72146df2/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/668f87b55b59a0c9f30e46ab843292e9b5112360/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/18e23be78c6bcd55863bf68b5c0a08cfb984a6e8/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/5967a02fd79bb95c77e29aba52b3830c778468a9/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/c9adba89c3f64817c4320128101b0596f0bec819/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/7b7120994be176894e3e56a9992ff493be7ab30f/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/4a40186d6d5bd8c4952093986197c331347acf69/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/3d0d2d13810fef50d6ebd5629e7f1646d097e810/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/4c62f7e43baae47b9d5969418ff9f0b3ea66fa54/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/4f3229309c0e56d2aebb133f9171ab3186de92b7/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/b55b19dd2febb3465d8be0758bdbdb91dcaacd9a/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/5d01f46577a9ba189e422adf12c27c38fa9db988/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/8ac327461d947b96a309eeb9e4510cc114025fdb/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/6daf6b51f2f6dd4cfe3320fa84257e00e460b68c/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/b355e20add9e37d2d4a11fd10bf917005d1f9040/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/c8b279317866ab9dbfdf99a5e417bebe072b63c8/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/d766c20d70195ddb9bd5f893c9474513fe299cb8/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/18f572a9d4db72411d414acb71ac2aa7c0860920/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/a7e14bfb880029fc1fb45528ab1a755b236b97de/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/f5c2ef850d62806c6c8826688a9714c2e8963f7b/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/c0196c3d37a68911b50c073eda647c82766a8b62/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/7f0b6c0ce0ec670b7181aa94004f8c8214facc3d/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/876b8713c518424b257f3c1b0fd2d6d9839148b1/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/43ce1f22a7fc6d438692b55577b6a0f2a7de46ee/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/47af43df271483e4459d35102e5b50c5/raw/3d97231242dd4e6f6f05fb37275771fe956028e4/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by yosefe** [comment]
bot:pipe:retest

**Comment by yosefe** [comment]
can't cd to /net/hpc-master//scrap/jenkins/workspace/ucx-2
bot:pipe:retest

**Pull request merged**